compute_endpoint/globus_compute_endpoint/endpoint/config/config.py

```diff
@@ -1,221 +1,220 @@
+        the endpoint connects to production.
+
+    :param local_compute_services: Point the endpoint to a local instance of the Compute
+        services (for Compute developers).
+
+    :param debug: If set emit debug-level log messages.
 
         This is a configuration implementation of the CLI's ``--debug`` flag.  Note
         that if this value is explicitly False, then the CLI flag, if utilized, will
         still put the EP into "debug mode."  The CLI wins.
-        Default: False
+    """
+
+    def __init__(
+        self,
+        *,
+        multi_user: bool = False,
+        display_name: str | None = None,
+        allowed_functions: t.Iterable[UUID_LIKE_T] | None = None,
+        authentication_policy: UUID_LIKE_T | None = None,
+        subscription_id: UUID_LIKE_T | None = None,
+        amqp_port: int | None = None,
+        heartbeat_period: float | int = 30,
+        environment: str | None = None,
+        local_compute_services: bool = False,
+        debug: bool = False,
+    ):
+        # Misc
+        self.display_name = display_name
+        self.debug = debug is True
+        self.multi_user = multi_user is True
+
+        # Connection info and tuning
+        self.amqp_port = amqp_port
+        self.heartbeat_period = heartbeat_period
+        self.environment = environment
+        self.local_compute_services = local_compute_services is True
 
-    detach_endpoint : Bool
-        Should the endpoint daemon be run as a detached process? This is good for
-        a real edge node, but an anti-pattern for kubernetes pods
-        Default: True
+        # Auth
+        self.allowed_functions = allowed_functions
+        self.authentication_policy = authentication_policy
+        self.subscription_id = subscription_id
 
-    log_dir : str
-        Optional path string to the top-level directory where logs should be written to.
-        Default: None
+        # Used to store the raw content of the YAML or Python config file
+        self.source_content: str | None = None
 
-    multi_user : bool | None
-        Designates the endpoint as a multi-user endpoint
-        Default: None
+    def __repr__(self) -> str:
+        kwds: dict[str, t.Any] = {}
+        for cls in type(self).__mro__:
+            fargspec = inspect.getfullargspec(cls.__init__)  # type: ignore[misc]
+            kwdefs = fargspec.kwonlydefaults
+            for kw in fargspec.kwonlyargs:
+                curval = getattr(self, kw)
+                if kwdefs and curval != kwdefs.get(kw):
+                    kwds.setdefault(kw, curval)
+
+        self_name = type(self).__name__
+        kwargs = (f"{k}={repr(v)}" for k, v in sorted(kwds.items()))
+        return f"{self_name}({', '.join(kwargs)})"
+
+    @property
+    def heartbeat_period(self):
+        return self._heartbeat_period
+
+    @heartbeat_period.setter
+    def heartbeat_period(self, val: float | int):
+        self._heartbeat_period = max(MINIMUM_HEARTBEAT, val)
```

Comment:
We probably should avoid changes that could modify EP behavior without the user knowing. However, this is pretty minor, and is already enforced in the `EndpointManager`, so maybe just add a warning log when the value is below the minimum.